File: frontend/pathipan-post-crud-nextjs/models/post.ts
// import mongoose, { Schema } from "mongoose";

// Define the TypeScript interface for a Post
// This interface describes the structure of a Post object
export type Post = {
  id: string;
  title: string;
  image: string;
  content: string;
};

// Define the schema for the Post model
// This schema defines the structure of the Post documents in the MongoDB collection
// const postSchema = new Schema({
//     title: { type: String, required: true },
//     image: { type: String, required: true },
//     content: { type: String, required: true },
// },
// {
//     timestamps: true,}
// )

// This will create a Post model if it doesn't already exist
// If it does exist, it will use the existing model
// const Post = mongoose.models.Post || mongoose.model("Post", postSchema);
// export default Post;

File: backend/golang/models/post.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Post struct {
	ID        primitive.ObjectID `json:"id" bson:"_id,omitempty" required:"true"`
	Title     string             `json:"title" bson:"title" required:"true"`
	Image     string             `json:"image" bson:"image" required:"true"`
	Content   string             `json:"content" bson:"content" required:"true"`
	CreatedAt time.Time          `json:"createdAt,omitempty" bson:"createdAt,omitempty"`
	UpdatedAt time.Time          `json:"updatedAt,omitempty" bson:"updatedAt,omitempty"`
}
